langchain/llms/azureml_endpoint.py

```diff
@@ -0,0 +1,255 @@
+
+    http_client: Any = None  #: :meta private:
+
+    content_formatter: Any = None
+    """The body handler class that provides an input and output
+    transform function to handle formats between the LLM and
+    the endpoint"""
+
+    """
+        Example:
+            .. code-block:: python
+
+            from langchain.llms.azureml_endpoint import LLMBodyHandler
+
+            class BodyHandler(LLMBodyHandler):
+                content_type = "application/json"
+                accepts = "application/json"
+                
+                def format_request_payload(self, prompt, model_kwargs) -> bytes:
+                    input_str = json.dumps({"inputs": {"input_string": [prompt]}, "parameters": model_kwargs})
+                    return str.encode(input_str)
+
+                def format_response_payload(self, output) -> str:
+                    response_json = json.loads(output)
+                    return response_json[0]["0"]
+                    
+    """
+
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+
+    @validator("http_client", always=True, allow_reuse=True)
+    @classmethod
+    def validate_client(cls, field_value: Any, values: Dict) -> AzureMLEndpointClient:
+        """Validate that api key and python package exists in environment."""
+        endpoint_key = get_from_dict_or_env(
+            values, "endpoint_api_key", "ENDPOINT_API_KEY"
+        )
+        endpoint_url = values["endpoint_url"]
+        deployment_name = values["deployment_name"]
+        http_client = AzureMLEndpointClient(endpoint_url, endpoint_key, deployment_name)
+        return http_client
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        _model_kwargs = self.model_kwargs or {}
+        return {
+            **{"deployment_name": self.deployment_name},
+            **{"model_kwargs": _model_kwargs},
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "azureml_endpoint"
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+    ) -> str:
+        """Call out to an AzureML Managed Online endpoint.
+
+        Args:
+            prompt: The prompt to pass into the model.
+            stop: Optional list of stop words to use when generating.
+
+        Returns:
+            The string generated by the model.
+
+        Example:
+            .. code-block:: python
+
+                response = azureml_model("Tell me a joke.")
+        """
+        _model_kwargs = self.model_kwargs or {}
+
+        body = self.content_formatter.format_request_payload(prompt, _model_kwargs)
+        endpoint_response = self.http_client.call(body)
+        response = self.content_formatter.format_response_payload(endpoint_response)
+        # TODO: Add error handling
```

Comment:
is it needed or does LLMBase already handles it?

Reply:
Errors are handled in `AzureMLEndpointClient.call` when invoking the endpoint and can be implemented in the content formatter as well when transforming content.